### PR TITLE
[feat]: [CDS-2949]: Rollback on approval step workflow.

### PR DESCRIPTION
**Pull request by lasdox**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/33610)

### CODE REVIEW

File: 400-rest/src/main/java/software/wings/service/impl/WorkflowExecutionServiceImpl.java [hunk 1]
```diff
@@ -617,26 +618,27 @@ public boolean updateNotes(String appId, String workflowExecutionId, ExecutionAr
...
     if (isNotEmpty(userGroupIds) && !verifyAuthorizedToAcceptOrReject(userGroupIds, appId, null)) {
       throw new InvalidRequestException("User not authorized to accept or reject the approval");
     }
 
-    return approveOrRejectExecution(appId, approvalDetails);
+    return approveOrRejectExecution(appId, approvalDetails, executionUuid);
   }
 
-  private boolean approveOrRejectExecution(String appId, ApprovalDetails approvalDetails) {
+  private boolean approveOrRejectExecution(String appId, ApprovalDetails approvalDetails, String executionUuid) {
```
Comment (Fernando-Dourado) [inline 400-rest/src/main/java/software/wings/service/impl/WorkflowExecutionServiceImpl.java:641]: An alternative could be to override the method avoiding others callers using null when they don't have an executionUuid. That creates a more clean code.

**Comment by abhinavmittal-harness** [comment]
*[`400-rest/src/main/java/software/wings/service/impl/WorkflowExecutionServiceImpl.java` line 678 at r2](https://reviewable.io/reviews/harness/harness-core/33610#-N3brPFaBgiSn9QoeoIY:-N3brPFb6t9dNJgD-Tag:b-qrnkh2) ([raw file](https://github.com/harness/harness-core/blob/12adbb705df935de22805f0c872785579cf40189/400-rest/src/main/java/software/wings/service/impl/WorkflowExecutionServiceImpl.java#L678)):*
> ```Java
>         triggerExecutionInterrupt(executionInterrupt);
>       }
> ```

We also wanted to do abort correct?

**Review by abhinavmittal-harness — APPROVED** [review]
> Reviewed 9 of 9 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33610)* status: :shipit: complete! ✓

**Comment by lasdox** [comment]
> _[`400-rest/src/main/java/software/wings/service/impl/WorkflowExecutionServiceImpl.java` line 678 at r2](https://reviewable.io/reviews/harness/harness-core/33610#-N3brPFaBgiSn9QoeoIY:-N3brPFb6t9dNJgD-Tag:b-qrnkh2) ([raw file](https://github.com/harness/harness-core/blob/12adbb705df935de22805f0c872785579cf40189/400-rest/src/main/java/software/wings/service/impl/WorkflowExecutionServiceImpl.java#L678)):_
> 
> > ```java
> >         triggerExecutionInterrupt(executionInterrupt);
> >       }
> > ```
> 
> We also wanted to do abort correct?

No need anymore if I understand product specs correctly

File: 400-rest/src/main/java/software/wings/service/impl/WorkflowExecutionServiceImpl.java (hunk 1 above)
Comment (lasdox) [reply 400-rest/src/main/java/software/wings/service/impl/WorkflowExecutionServiceImpl.java:641]: Dont know why I didnt think about this. Nice suggestion, thanks man!

**Comment by raghuAtWings** [comment]
*[`957-cg-beans/src/main/java/io/harness/beans/ExecutionInterruptType.java` line 73 at r6](https://reviewable.io/reviews/harness/harness-core/33610#-N5a3-Up9--OnX-LSgDn:-N5a3-Up9--OnX-LSgDo:b-8cpwkh) ([raw file](https://github.com/harness/harness-core/blob/89541542bf0df8064d7f38c919f5e5f142fd227a/957-cg-beans/src/main/java/io/harness/beans/ExecutionInterruptType.java#L73)):*
> ```Java
> 
>   ROLLBACK_PROVISIONER_AFTER_PHASES("Rollback Provisioner after Phases"),
>   ROLLBACK_ON_APPROVAL("Rollback on approval"),
> ```

the new enum should always be in the end

**Review by raghuAtWings — COMMENTED** [review]
> Reviewed 1 of 9 files at r2, 1 of 3 files at r3, 6 of 6 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33610)* status: 2. unresolved discussions. 0 of 1 approvals obtained.

**Review by raghuAtWings — APPROVED** [review]
> Reviewed 3 of 3 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33610)* status: 2. unresolved discussions.

**Review by abhinavmittal-harness — COMMENTED** [review]
> Reviewed 1 of 3 files at r3, 5 of 6 files at r6, 3 of 3 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33610)* status: 2. unresolved discussions.

**Review by abhinavmittal-harness — DISMISSED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33610)* status: 1. unresolved discussions.

**Review by lasdox — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33610)* status: 1. unresolved discussions. 

___
*[`400-rest/src/main/java/software/wings/service/impl/WorkflowExecutionServiceImpl.java` line 641 at r2](https://reviewable.io/reviews/harness/harness-core/33610#-N31tJdpa34Q0tK4lRU1-r2-641:-N5ebrnB4123KvJdxpsN:b32w0x) ([raw file](https://github.com/harness/harness-core/blob/12adbb705df935de22805f0c872785579cf40189/400-rest/src/main/java/software/wings/service/impl/WorkflowExecutionServiceImpl.java#L641)):*
<details><summary><i>Previously, lasdox (Lucas Sales) wrote…</i></summary><blockquote>

Dont know why I didnt think about this. Nice suggestion, thanks man!
</blockquote></details>

OK

**Review by Fernando-Dourado — APPROVED** [review]
> Reviewed 1 of 9 files at r2, 1 of 3 files at r3, 5 of 6 files at r6, 1 of 3 files at r7, 2 of 2 files at r8, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33610)* status: :shipit: complete! ✓

**Review by Fernando-Dourado — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33610)* status: :shipit: complete! ✓

**Pull request merged**